api/v1beta1/common_webhook.go

```diff
@@ -28,3 +33,42 @@ const (
 		"ensures that the copying of the logs to the PV is completed without any " +
 		"complications."
 )
+
+// merge non-workflow section into workflow
+func mergeSectionIntoWorkflow(instance interface{}, workflowStepNum int) {
+        spec, ok := instance.(*TempestSpec)
+        if !ok {
+        fmt.Println("Error, instance is not of type *TempestSpec")
+                return
+        }
+
+        tRun := spec.TempestRun
+        wtRun := &spec.Workflow[workflowStepNum].TempestRun
+
+        tRunReflect := reflect.ValueOf(tRun)
+        wtRunReflect := reflect.ValueOf(wtRun).Elem()
+
+	setNonZeroValues(tRunReflect, wtRunReflect, false)
+}
+
+func setNonZeroValues(src reflect.Value, dest reflect.Value, is_struct bool) {
+        for i := 0; i < src.NumField(); i++ {
+                tRunName := src.Type().Field(i).Name
+                tRunValue := src.Field(i)
+                wtRunValue := dest.FieldByName(tRunName)
+
+                if wtRunValue.IsZero() && !tRunValue.IsZero() {
```

Comment:
Nice catch, I think the need for `!tRunValue.IsNil` is still there though :thinking: The problem was that if the value in tempestRun was nil, it would create an error.

Reply:
Yes, you are right:

```
The Tempest "tempest-tests" is invalid:
* spec.workflow[0].tempestRun.extraImagesType: Invalid value: "null": spec.workflow[0].tempestRun.extraImagesType in body must be of type array: "null"
* spec.workflow[0].tempestRun.externalPlugin: Invalid value: "null": spec.workflow[0].tempestRun.externalPlugin in body must be of type array: "null"
* spec.workflow[0].tempestRun.extraRPMs: Invalid value: "null": spec.workflow[0].tempestRun.extraRPMs in body must be of type array: "null"
```

Notice that the error only impacts arrays. We are trying to assign null to what should be a pointer to an array. Is there a way how to fix this? What about checking whether the array is empty in the tRun section and if it is then do not perform the assignment?